src/rcc.rs

```diff
@@ -159,7 +168,20 @@ mod usb_clocking {
     }
 }
 
+/// Backup Domain Control register (RCC_BDCR)
+pub struct BDCR {
+    _0: (),
+}
+
+impl BDCR {
+    pub(crate) fn bdcr(&mut self) -> &rcc::BDCR {
+        // NOTE(unsafe) this proxy grants exclusive access to this register
+        unsafe { &(*RCC::ptr()).bdcr }
+    }
+}
+
 /// Clock configuration
+#[derive(Clone, Copy)]
```

Comment:
We must not allow cloning of the `CFGR` struct. This struct is responsible for making sure that the clocks are only configured once. When you call `freeze()` it consumes the struct and thereby ensures the clocks won't change anymore, so other code can rely on that. If you make `CFGR` clonable you could suddenly have multiple instances of this struct around, which would allow you to change the clocks and break the assumption that they are stable